client/src/pages/create-app.tsx

```diff
@@ -23,81 +55,205 @@ export const CreateApp = () => {
-            name: values.name,
-          },
-          refetchQueries: [
-            {
-              query: DashboardDocument,
+            input: {
+              name: values.name,
+              gitRepoUrl: values.gitRepoUrl,
+              branchName: values.gitBranch ? values.gitBranch : undefined,
             },
-          ],
+          },
         });
-        if (data?.data) {
-          toast.success('App created successfully');
-          history.push(`/app/${data.data.createApp.app.id}`);
-        }
+        setIsTerminalVisible(true);
       } catch (error) {
         toast.error(error.message);
       }
     },
   });
 
+  const handleNext = () => {
+    setIsTerminalVisible(false);
+    const appId = arrayOfCreateAppLogs[arrayOfCreateAppLogs.length - 1].message;
+    history.push(`app/${appId}`);
+  };
+
+  // Effect for app creation
+  useEffect(() => {
+    isAppCreationSuccess === AppCreationStatus.FAILURE
+      ? toast.error('Failed to create an app')
+      : isAppCreationSuccess === AppCreationStatus.SUCCESS &&
+        toast.success('App created successfully');
+  }, [isAppCreationSuccess]);
+
   return (
     <>
       <Header />
 
       <div className="max-w-5xl mx-auto py-6 px-4 sm:px-6 lg:px-8">
         <h1 className="text-lg font-bold">Create a new app</h1>
-        <div className="mt-4 mb-4">
-          <h2 className="text-gray-400">
-            Enter app name, click create and voila!
-          </h2>
-        </div>
-        <form>
-          <div className="grid grid-cols-1 md:grid-cols-3">
-            <div>
-              <FormLabel>App name: </FormLabel>
-              <FormInput
-                autoComplete="off"
-                id="name"
-                name="name"
-                placeholder="Name"
-                value={formik.values.name}
-                onChange={formik.handleChange}
-                onBlur={formik.handleBlur}
-                error={Boolean(formik.errors.name && formik.touched.name)}
-              />
-              {formik.errors.name ? (
-                <FormHelper status="error">{formik.errors.name}</FormHelper>
-              ) : null}
+        <div className="mt-12">
+          {isTerminalVisible ? (
+            <>
+              <p className="mb-2 ">
+                Creating <b>{formik.values.name}</b> app from{' '}
+                <b>{formik.values.gitRepoUrl}</b>
+              </p>
+              <p className="text-gray-500 mb-2">
+                Creating app usually takes a couple of minutes. Breathe in,
+                breathe out, logs are about to appear below:
+              </p>
+              <Terminal className={'w-6/6'}>
+                {console.log(arrayOfCreateAppLogs)}
+                {arrayOfCreateAppLogs.map((log) => (
+                  <p
+                    key={arrayOfCreateAppLogs.indexOf(log)}
+                    className={'text-s leading-5'}
+                  >
+                    {log.message?.replaceAll('[1G', '')}
```

Comment:
@pradel this `replaceAll` thing seems pretty bad, but it works. Using the same on the server side before sending the log somehow was not working.